components/containerd/pod/component.go

```diff
@@ -57,6 +58,17 @@ func (c *component) States(ctx context.Context) ([]components.State, error) {
 		return nil, err
 	}
 	if last.Error != nil {
+		lastReboot, err := reboot.LastReboot(context.Background())
+		if err == nil && time.Since(lastReboot) < initializeGracePeriod {
```

Comment:
Is it possible to check this condition in the state collection handler? And if this condition is true, tell the controle-plane: `I am initializing, please wait a moment`?

Reply:
yea thats doable..but will affect all components (all components will be default initializing when machine just boot up)..if this checking is local to each component is more flexible..

Reply:
In your implementation, only check if the duration from last reboot is less than initialize grace period when the component is unhealthy. We can do same thing in the state collection handler...